src/moj/components/date-picker/date-picker.js

```diff
@@ -0,0 +1,596 @@
+/**
+ * Datepicker config
+ *
+ * @typedef {object} DatepickerConfig
+ *
+ * @property {string} [imagePath] - The path to image assets.
+ * @property {string} [id] - .
+ * @property {string} [name] - .
+ * @property {string} [label] - .
+ * @property {string} [hint] - .
+ * @property {string} [minDate] - .
+ * @property {string} [maxDate] - .
+ */
+
+/**
+ * Datepicker component
+ *
+ * @param {HTMLElement} $module - HTML element
+ * @param {DatepickerConfig} config - Datepicker config
+ * @constructor
+ */
+function Datepicker($module, config) {
+  if (!$module) {
+    return this
+  }
+  const defaultConfig = {
+    imagePath: '/assets/images/',
+  }
+  this.config = { ...defaultConfig, ...config }
+
+  this.dayLabels = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday']
+  this.monthLabels = [
+    'January',
+    'February',
+    'March',
+    'April',
+    'May',
+    'June',
+    'July',
+    'August',
+    'September',
+    'October',
+    'November',
+    'December',
+  ]
+
+  this.currentDate = new Date()
+  this.currentDate.setHours(0, 0, 0, 0)
+  this.calendarDays = []
+
+  this.keycodes = {
+    tab: 9,
+    esc: 27,
+    pageup: 33,
+    pagedown: 34,
+    end: 35,
+    home: 36,
+    left: 37,
+    up: 38,
+    right: 39,
+    down: 40,
+  }
+
+  this.$module = $module
+  this.$input = $module.querySelector('.moj-js-datepicker-input')
+  this.$calendarButton = $module.querySelector('.moj-js-datepicker-button')
+}
+
+/**
+ * Initialise Datepicker
+ */
+Datepicker.prototype.init = function () {
+  // Check that required elements are present
+  if (!this.$input) {
+    return
+  }
+
+  this.initControls()
+}
+
+/**
+ * Initialise controls and set attributes
+ */
+Datepicker.prototype.initControls = function () {
+  // Create datepicker popup dialog
+  const titleId = `datepicker-title-${this.$input.id}`
+  const dialog = document.createElement('div')
+  dialog.id = `datepicker-${this.$input.id}`
+  dialog.setAttribute('class', 'moj-datepicker__dialog  datepickerDialog')
+  dialog.setAttribute('role', 'dialog')
+  dialog.setAttribute('aria-modal', 'true')
+  dialog.setAttribute('aria-labelledby', titleId)
+  dialog.innerHTML = this.createDialogMarkup(titleId)
+
+  this.dialogElement = dialog
+  this.$input.insertAdjacentElement('afterend', this.dialogElement)
+
+  this.dialogTitleNode = this.dialogElement.querySelector('.js-datepicker-month-year')
+
+  this.setMinAndMaxDatesOnCalendar()
+
+  // create calendar
+  const tbody = this.dialogElement.querySelector('tbody')
+  let dayCount = 0
+  for (let i = 0; i < 6; i++) {
+    // create row
+    const row = tbody.insertRow(i)
+
+    for (let j = 0; j < 7; j++) {
+      // create cell (day)
+      const cell = document.createElement('td')
+      const dateButton = document.createElement('button')
+      dateButton.dataset.form = 'date-select'
```

Comment:
I'm not sure what `.dataset.form` is doing here or if it's needed.

We might need to add `aria-label` to explain what the button does (e.g. "28th January 2024"), but I'm not sure what state of the art is here